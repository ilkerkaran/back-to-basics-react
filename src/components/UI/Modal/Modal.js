import React, { PureComponent } from 'react';

import './Modal.css';
import PropTypes from 'prop-types';
import Backdrop from '../Backdrop/Backdrop';

class Modal extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <>
        <Backdrop
          show={this.props.show}
          onClick={this.props.onClose}
        />
        <div
          className="Modal"
          style={{
            transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
            opacity: this.props.show ? '1' : '0'
          }}
        >
          {this.props.children}
        </div>
      </>
    );
  }
}
export default Modal;

Modal.propTypes = {
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
};
