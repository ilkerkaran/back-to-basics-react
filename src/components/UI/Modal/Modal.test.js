import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Modal from './Modal';


describe('<Modal /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Modal show={true} onClose={() => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Modal /> ', () => {
  it(' renders div witha a class name Modal', () => {
    const spinner = shallow(<Modal show onClose={() => {}}/>);
    expect(spinner.find('.Modal')).toHaveLength(1);
  });
});
