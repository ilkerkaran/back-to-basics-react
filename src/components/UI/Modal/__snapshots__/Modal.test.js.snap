// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> snapshot  renders correctly 1`] = `
Array [
  <div
    className="Backdrop"
    onClick={[Function]}
  >
     
  </div>,
  <div
    className="Modal"
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  />,
]
`;
